test/base/test_store.py

```diff
@@ -141,3 +145,83 @@ def test_storage_trace():
                                 getattr(mem_res['history'],
                                         f'get_{entry}_trace')()[iteration],
                                 hdf5_entry_trace[iteration])
+
+
+def test_storage_profiling():
```

Comment:
for (new/all) tests please a one-line """This test tests this and that stuff""" docstring at the beginning. (or a longer description too if necessary)

Reply:
this would allow noticing when e.g. some variables are not saved for whatever reason